worker/dependency/util.go

```diff
@@ -100,3 +101,20 @@ func SelfManifold(engine *Engine) Manifold {
 		},
 	}
 }
+
+// ManifoldDependencies returns all manifold dependencies.
+func (all Manifolds) ManifoldDependencies(name string, manifold Manifold) set.Strings {
+	if len(manifold.Inputs) == 0 {
+		return set.NewStrings(name)
+	}
+	result := set.NewStrings()
+	for _, input := range manifold.Inputs {
+		result.Add(input)
+		deps := all.ManifoldDependencies(input, all[input])
+		for _, v := range deps.SortedValues() {
+			result.Add(v)
+		}
+	}
+	result.Remove("")
```

Comment:
Why's this here? (I mean, in a deeper sense than because there's an extra "" in the results otherwise.) Worth a comment if it's truly necessary.